contracts/Router.sol

```diff
@@ -0,0 +1,188 @@
+            c.amountsPledged[msg.sender] = c.amountsPledged[msg.sender].add(_amount);
+        }
+
+        emit Pledge(numCampaigns, msg.sender, _amount);
+    }
+
+
+    function unpledge(uint256 _amount, bool max) external payable {
+        Campaign storage c = campaigns[numCampaigns];
+
+        uint256 pledgerId = getPledgerId(numCampaigns, msg.sender);
+        require(pledgerId != 0, "User has not pledged.");
+        if (max) {_amount = c.amountsPledged[msg.sender];}
+        require(_amount <= c.amountsPledged[msg.sender], "Cannot unpledge more than already pledged.");
+
+        c.totalPledged = c.totalPledged.sub(_amount);
+        c.amountsPledged[msg.sender] = c.amountsPledged.sub(_amount);
+
+        reserveToken.transfer(msg.sender, _amount);
+
+        emit Unpledge(numCampaigns, msg.sender, _amount);
+    }
+
+
+    function swapExactTokensForTokens(
+        uint amountIn,
+        uint amountOutMin,
+        address[] calldata path,
+        address to,
+        uint deadline
+    ) external override returns (uint256[] memory amounts) {
+        require(deadline >= block.timestamp, 'UniswapV2Router: EXPIRED');
+
+        Campaign storage c = campaigns[numCampaigns];
+        require(c.totalPledged >= amountIn, "Not enough DAI pledged. Rebase postponed.");
+
+        // transfer pledged reserveToken to reserve
+        reserveToken.increaseAllowance(address(this), c.totalPledged);
+        reserveToken.transfer(RESERVE, c.totalPledged);
+
+        // Send TREE to each pledger
+        for (uint i=1; i<c.numPledgers+1; i++) {
+            
+            address pledger = c.pledgers[i];
+            uint256 amountPledged = c.amountsPledged[pledger];
+
+            // treeToReceive = value pledged * (amountIn / totalPledged)
+            // For example, if 100 DAI is pledged and there's only 50 TREE available
+            // an address that pledged 5 DAI would receive 5 * (50/100) = 2.5 TREE
+            uint256 treeToReceive = amountPledged.mul(amountIn).div(c.totalPledged);
+
+            // Only transfer to EOAs to prevent unexpected reverts if pledge was done using CREATE2
+            // Also, if user ended up unpledging 100%, do not waste a transfer of 0 tokens
+            // note: TREE is already approved to transfer
+            // https://github.com/WhalerDAO/tree-contracts/blob/4525d20def8fce41985f0711e9b742a0f3c0d30b/contracts/TREEReserve.sol#L228
+            if (!Address.isContract(pledger) && amountPledged > 0) {
+                tree.transfer(pledger, treeToReceive);
+                c.treeSold = c.treeSold + treeToReceive;
+            }
+        }
+        numCampaigns++;
+
+        // Return amounts based on https://github.com/WhalerDAO/tree-contracts/blob/4525d20def8fce41985f0711e9b742a0f3c0d30b/contracts/TREEReserve.sol#L217
+        uint256[] memory amounts = new uint256[](2);
+        amounts[0] = c.treeSold;
+        amounts[1] = c.totalPledged;
```

Comment:
`amounts[0]` is never used. (See https://github.com/WhalerDAO/tree-contracts/blob/master/contracts/TREEReserve.sol#L239 )
`amounts[1]` should be `0` since this contract is keeping all the reserve tokens
```suggestion
```

Reply:
Ah okay, I figured as this contract was used as a router that it wouldn't be holding funds.  Thanks for clarifying